flutter/pubspec.yaml

```diff
@@ -13,10 +13,7 @@ dependencies:
     sdk: flutter
   flutter_web_plugins:
     sdk: flutter
-  sentry: #">=4.0.0-beta.2 <4.1.0"
-    # TODO: we need the specific version for publishing it, but the relative path for development.
-    # Figure out how to do it via Craft for releases.
-    path: ../dart
+  sentry: ">=4.0.0-beta.2 <4.1.0"
```

Comment:
```suggestion
  sentry: ">=4.0.0-beta.1 <4.1.0"
```

Reply:
This is not needed, it is overridden, the error is not coming from here.

Reply:
even better, thanks.

Reply:
Even with this change pana fails locally, doesn't seem to respect pre-release versions.
```
The constraint `>=4.0.0-beta.1 <4.1.0` on sentry does not support the latest published version `3.0.1`
```

Reply:
is it possible to suppress this error somehow? not a pana expert, otherwise I'll search for it tomorrow.

Reply:
Just looked a bit but there doesn't seem to be a way that would just suppress this and not everything else.

Reply:
damn, looks like a valid use-case, maybe it's worth opening a new issue on pana's repo.
https://github.com/dart-lang/pana

Reply:
There is this https://github.com/dart-lang/pana/issues/242

Reply:
The overrides are being stripped before analysis so I don't see a way to work around this.

Reply:
that makes sense I'd say, but the suppress here could just be able to use nonstable versions as alpha, beta, RCs, right.

> The constraint `>=4.0.0-beta.1 <4.1.0` on sentry does not support the latest published version `3.0.1`

if we are able to suppress only this, it'd be fine I guess

Reply:
This is done via the https://github.com/axel-op/dart-package-analyzer action. Could probably exclude the up-to-date dependency score here: https://github.com/getsentry/sentry-dart/blob/main/.github/workflows/flutter.yml#L112

Reply:
yeah, it's probably running `pub outdated` with `--no-prereleases`.

but it seems that `pub upgrade` fails in the GH action, so I don't see how excluding the up-to-date dependencies would make the GH Action happy, looks like they'd need to offer a way to toggle this on/off.

Reply:
nvm, I could reproduce it locally, subtracting 10 points would work, let's just do this and add a comment, so I suggest:

```
# 90 due to 'pana' dependencies score when using 'dependency_overrides'
if (( $PERCENTAGE < 90 ))
```

could you do this change and see if CI gets happy? thanks :)

Reply:
Sure, that was my idea as well, don't see any other way atm.

Reply:
.... It runs `pub outdated --no-dev-dependencies --up-to-date --no-dependency-overrides` so this just completely errors.

Reply:
I don't think there currently is a way of running pana without setting everything back to `3.0.1` and only updating the version prior to release.

Reply:
cool, I think we still need to use `beta.1` here or it fails as it does not resolve the dependency, if we do `beta.1`, it fails because of API incompatibility but this is fine as I'm releasing a beta.2 today :)

Reply:
> cool, I think we still need to use beta.1 here or it fails as it does not resolve the dependency, if we do beta.1, it fails because of API incompatibility but this is fine as I'm releasing a beta.2 today :)

It will always fail this way due to the command using `--no-dev-dependencies` and `--no-dependency-overrides`.
I am not sure what to do here. This pana error is not valid as soon as you release it so it should somehow be ignored.

Reply:
can we try using `beta.1`? if it fails because of API incompatibility, I don't see why this is a problem.

Reply:
It doesn't find `beta.1` due to `--no-dev-dependencies`.

Reply:
mm that's weird, today morning I opened https://github.com/getsentry/sentry-dart/pull/233/files and tried using beta.1 and it worked out, it failed because of breaking changes, but not other reasons.

Reply:
tbh we are running other CI jobs with `format`, `analyze` and `tests`, this is just nice to have, I consider the outcome of this PR more important than the score check.

Reply:
last case, it'd be fine to deactivate the GH action that runs `pana` for now and enable it back once both packages are stable.

Reply:
pana doesn't seem to take a mono repo with 2 or more dependent packages into account, this will pretty much always fail for dev versions until this is changed.

Reply:
if it depends on a stable package and API is compatible, it's alright I guess, at least I've seen other packages using this GH Action too.
So here it is #235
Once this is merged, it'd be fine to just fix the conflicts